File: 0252-meeting-rooms/0252-meeting-rooms.cpp
class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
        if(intervals.empty()) return true; 
        sort(intervals.begin(), intervals.end(),[](vector<int>& a, vector<int>& b){
            return a[0] < b[0]; 
        });

        int currEnd = intervals[0][1]; 

        for(int i = 1; i < intervals.size(); i++){
            int nextStart = intervals[i][0]; 
            int nextEnd = intervals[i][1]; 

            if(nextStart < currEnd) return false; 

            currEnd = nextEnd; 
        }

        return true; 
    }
};

File: 0252-meeting-rooms/0252-meeting-rooms.java
class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        if(intervals.length == 0) return true; 
        Arrays.sort(intervals, (a,b) -> Integer.compare(a[0],b[0])); 
        int endTime = intervals[0][1]; 
        for(int i = 1; i < intervals.length; i++){
            int nextEndTime = intervals[i][1]; 
            int nextStartTime = intervals[i][0]; 
            if(endTime > nextStartTime) return false; 
            endTime = nextEndTime; 
        }
        return true; 
    }
}